modules/javafx.graphics/src/main/java/javafx/scene/robot/Robot.java

```diff
@@ -0,0 +1,386 @@
+     * other than the JavaFX Application Thread.
+     * @throws NullPointerException if buttons is {@literal null}.
+     */
+    public void mouseRelease(MouseButton... buttons) {
+        Objects.requireNonNull(buttons, "buttons must not be null");
+        peer.mouseRelease(buttons);
+    }
+
+    /**
+     * Clicks the specified {@link MouseButton}s.
+     *
+     * @implSpec This is a convenience method that is equivalent to calling
+     * {@link #mousePress(MouseButton...)} followed by {@link #mouseRelease(MouseButton...)}.
+     * @param buttons the mouse buttons to click
+     * @throws IllegalStateException if this method is called on a thread
+     * other than the JavaFX Application Thread.
+     * @throws NullPointerException if buttons is {@literal null}.
+     */
+    public void mouseClick(MouseButton... buttons) {
+        Objects.requireNonNull(buttons, "buttons must not be null");
+        mousePress(buttons);
+        mouseRelease(buttons);
+    }
+
+    /**
+     * Scrolls the mouse wheel by the specified amount of wheel clicks. Positive
+     * {@code wheelAmt} scrolls up whereas negative scrolls down.
+     *
+     * @param wheelAmt the (signed) amount of clicks to scroll the wheel
+     * @throws IllegalStateException if this method is called on a thread
+     * other than the JavaFX Application Thread.
+     */
+    public void mouseWheel(int wheelAmt) {
+        peer.mouseWheel(wheelAmt);
+    }
+
+    /**
+     * Returns the {@link Color} of the pixel at the specified screen coordinates
+     * relative to the primary screen.
+     *
+     * @param x the x coordinate to get the pixel color from
+     * @param y the y coordinate to get the pixel color from
+     * @return the pixel color at the specified screen coordinates
+     * @throws IllegalStateException if this method is called on a thread
+     * other than the JavaFX Application Thread.
+     */
+    public Color getPixelColor(double x, double y) {
+        return peer.getPixelColor(x, y);
+    }
+
+    /**
+     * Returns the {@link Color} of the pixel at the screen coordinates relative to the
+     * primary screen specified by {@code location}.
+     *
+     * @param location the (x,y) coordinates to get the pixel color from
+     * @return the pixel color at the specified screen coordinates
+     * @throws NullPointerException if the given {@code location} is {@literal null}
+     * @throws IllegalStateException if this method is called on a thread
+     * other than the JavaFX Application Thread.
+     */
+    public Color getPixelColor(Point2D location) {
+        return getPixelColor(location.getX(), location.getY());
+    }
+
+    /**
+     * Returns a {@link WritableImage} containing the specified rectangular area relative
+     * to the primary screen. If the given {@code image} is {@literal null}, or if the given
+     * {@code image} is too small, a new {@code WritableImage} will be created and returned.
```

Comment:
will also create a new image if the image is too big, so maybe say:

`if the given image is not the required size`